impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -252,9 +260,14 @@
     id<MTLCommandBuffer> command_buffer =
         ContextMTL::Cast(context.get())
             ->CreateMTLCommandBuffer("Present Waiter Command Buffer");
-    [command_buffer commit];
-    [command_buffer waitUntilScheduled];
-    [drawable_ present];
+    if (ShouldPresentSync()) {
+      [command_buffer commit];
+      [command_buffer waitUntilScheduled];
+      [drawable_ present];
+    } else {
+      [command_buffer presentDrawable:drawable_];
+      [command_buffer commit];
+    }
```

Comment:
This should be safe if we're not presenting with transaction, but needs more testing from me.